### ISSUE TITLE
Bug: [space-infix-ops] Missing error report for conditional types

### ISSUE THREAD

**Issue by holazz** [typescript-eslint/typescript-eslint#5033]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.6.4&sourceType=module&code=C4TwDgpgBAKhDOwA8MB8UC8soQB7AgDsATeKAIwHtKAbCAQ0IH5gAnAVwgC4Azem+BABQQA&rules=N4IgAgLgngDgpgZwMYCcCWMIFpEBs0B2EA9AjAIZJxaEBmaAHlgPYwIgBcIcKKzKIAL5A&tsConfig=N4XyA

### Repro Code

```typescript
type Test<T> = T extends boolean?true:false
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/space-infix-ops": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

```ts
type Test<T> = T extends boolean ? true : false
```

### Actual Result

```ts
type Test<T> = T extends boolean?true:false
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.25.0` |
| `@typescript-eslint/parser`            | `5.25.0` |
| `TypeScript`                           | `4.6.4` |
| `ESLint`                               | `8.15.0` |
| `node`                                 | `16.15.0` |

**Comment by holazz** [comment]
fixed it  [here](https://github.com/typescript-eslint/typescript-eslint/compare/main...holazz:fix/space-infix-ops), waiting to be marked accepting PRs.
